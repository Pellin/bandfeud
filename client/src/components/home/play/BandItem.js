import React from 'react';
import Points from '../play/Points';

const BandItem = ({ name, url, points }) => {
  const fireReg = new RegExp('Firefox');
  const isFirefox = fireReg.test(navigator.userAgent);
  return (
    <div className="band-item">
      <div className="image-container">
        <div className="band-item-image">
          <img className="img" src={url} alt={'"' + { name } + '"'} />
        </div>
        {points && <Points points={points} />}
      </div>
      <div className={isFirefox ? 'title-flow-firefox' : 'title-flow'}>
        <div className="band-item-title-container">
          <div className="band-item-title">{name}</div>
        </div>
        
      </div>
    </div>
  );
};

export default BandItem;
