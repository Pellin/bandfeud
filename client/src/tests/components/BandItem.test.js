import React from 'react';
import { shallow } from 'enzyme';

import BandItem from '../../components/home/play/BandItem';

it('should render BandItem correctly', () => {
  const wrapper = shallow(<BandItem name="one" url="www.1img.com" />);

  expect(wrapper.find('.band-item-title').text()).toBe('one');
  expect(wrapper.find('.band-item-title').children().length).toBe(1);
  expect(wrapper.find('img').prop('src')).toBe('www.1img.com');
  expect(wrapper).toMatchSnapshot();
});
