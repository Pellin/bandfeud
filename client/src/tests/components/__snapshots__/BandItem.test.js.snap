// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render BandItem correctly 1`] = `
<div
  className="band-item"
>
  <div
    className="image-container"
  >
    <div
      className="band-item-image"
    >
      <img
        alt="\\"[object Object]\\""
        className="img"
        src="www.1img.com"
      />
    </div>
  </div>
  <div
    className="title-flow"
  >
    <div
      className="band-item-title-container"
    >
      <div
        className="band-item-title"
      >
        one
      </div>
    </div>
  </div>
</div>
`;
